File: backend/profiles_svc/app/schemas/collaborator.py
from __future__ import annotations

from typing import List, Optional

from sqlmodel import SQLModel

from app.models.collaborator_profile import WorkMode
from app.schemas.interest import InterestRead
from app.schemas.portfolio import PortfolioItemRead
from app.schemas.skill import SkillRead


class CollaboratorBase(SQLModel):
    headline: Optional[str] = None
    bio: Optional[str] = None
    location_city: Optional[str] = None
    location_state: Optional[str] = None
    availability_hours_per_week: Optional[int] = None
    work_mode: Optional[WorkMode] = None


class CollaboratorCreate(CollaboratorBase):
    pass


class CollaboratorUpdate(CollaboratorBase):
    pass


class CollaboratorRead(CollaboratorBase):
    id: int
    user_id: int
    skills: List[SkillRead] = []
    interests: List[InterestRead] = []
    portfolio_items: List[PortfolioItemRead] = []

    class Config:
        orm_mode = True

File: frontend/src/types/collaborator.ts
import { WorkMode } from "./enums";
import { ProjectTag } from "./project";

export interface Skill {
  id: number;
  name: string;
  level: number;
}

export interface Interest {
  id: number;
  slug: string;
  label: string;
}

export interface PortfolioItem {
  id: number;
  title: string;
  url: string;
  description?: string;
  thumb_url?: string;
}

export interface CollaboratorProfile {
  id: number;
  user_id: number;
  headline?: string;
  bio?: string;
  location_city?: string;
  location_state?: string;
  availability_hours_per_week?: number;
  work_mode?: WorkMode;
  skills: Skill[];
  interests: Interest[];
  portfolio_items: PortfolioItem[];
}

export interface CollaboratorPayload {
  headline?: string;
  bio?: string;
  location_city?: string;
  location_state?: string;
  availability_hours_per_week?: number;
  work_mode?: WorkMode;
}

export interface SkillPayload {
  name: string;
  level: number;
}

export interface InterestPayload {
  slug: ProjectTag;
  label: string;
}

export interface PortfolioPayload {
  title: string;
  url: string;
  description?: string;
  thumb_url?: string;
}

